src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -1140,13 +1140,15 @@ private void ThrowIfWebhooksNotConfigured()
 
         internal bool TryGetRpcBaseUrl(out Uri rpcBaseUrl)
         {
-            if (this.durableTaskOptions.LocalRpcEndpointEnabled != false)
+            if (this.durableTaskOptions.LocalRpcEndpointEnabled != false
+                && this.localHttpListener.IsListening)
             {
                 rpcBaseUrl = this.localHttpListener.InternalRpcUri;
                 return true;
             }
 
-            // The app owner explicitly disabled the local RPC endpoint.
+            // The app owner explicitly disabled the local RPC endpoint, or we were unable to
+            // successfully find a port after several tries.

```

Comment:
I don't see the "several tries" logic in here. If it's elsewhere, could we document that?

Reply:
I just removed that, because now this code shouldn't fail if we failed to start up, since we are throwing an exception.